src/server/pps/server/api_server.go

```diff
@@ -173,6 +179,19 @@ func validateNames(names map[string]bool, input *pps.Input) error {
 	return nil
 }
 
+func validateName(name string) error {
```

Comment:
We should probably have a unit test for the calling code; I looked for one but didn't find it.  (validateNames calls validateName, validateInput calls validateNames, validatePipeline calls validateInput; we should at least test validatePipeline or perhaps validateInput).

Reply:
Yeah, I think a test that at least ensures the validation is run probably makes sense. I'll add a TODO. I don't know if it really makes sense to verify all of the simpler validations though, like the names and whatnot.

Reply:
It's a shame that we wrote an integration test instead of a unit test, but no need to fix in this PR.